tests/cross/src/test/scala/tests/tokens/SemanticTokensSuite.scala

```diff
@@ -292,4 +292,33 @@ class SemanticTokensSuite extends BaseSemanticTokensSuite {
         |}
         |""".stripMargin,
   )
+
+  // When for-comprehension includes line with `=`, we get `scala.x$1`, `scala.x$2` symbols on `foo`.
+  // Both `scala` and `x$#` have position on `foo`, and we don't want to highlight it as a `scala` package,
+  // so we need `namespace` to have lower priority than `variable`.
+  check(
+    "for-comprehension",
+    s"""|package <<example>>/*namespace*/
+        |
+        |object <<B>>/*class*/ {
+        |  val <<a>>/*variable,definition,readonly*/ = for {
+        |    <<foo>>/*parameter,declaration,readonly*/ <- <<List>>/*class*/("a", "b", "c")
```

Comment:
```suggestion
        |    <<foo>>/*variable,declaration,readonly*/ <- <<List>>/*class*/("a", "b", "c")
```
Sorry for coming back to this so late, but this might end up confusing for users. The usage is no longer a parameter.

Reply:
I merged the other PR, we can rebase this one :fire: